### PR TITLE
Fix issues with the combination of position:sticky and tables

### LINKED ISSUE #18441: WebRender: assertion failed: !self.nodes.contains_key(&id)
100% reproducible when visiting http://eztv.ag

```gdb
assertion failed: !self.nodes.contains_key(&id) (thread RenderBackend, at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/clip_scroll_tree.rs:360)
stack backtrace:
   0:     0x557da3ee42e4 - backtrace::backtrace::libunwind::trace
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/libunwind.rs:53
                         - backtrace::backtrace::trace<closure>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/mod.rs:42
   1:     0x557da3ede86f - backtrace::capture::{{impl}}::new
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/capture.rs:64
   2:     0x557d9f611ccc - servo::main::{{closure}}
                        at /shared/dev/rust/servo/ports/servo/main.rs:130
   3:     0x557da68f8316 - std::panicking::rust_panic_with_hook
                        at /checkout/src/libstd/panicking.rs:612
   4:     0x557da0f911a6 - std::panicking::begin_panic<&str>
                        at /checkout/src/libstd/panicking.rs:572
   5:     0x557da1114603 - webrender::clip_scroll_tree::{{impl}}::add_node
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/clip_scroll_tree.rs:360
   6:     0x557da1114395 - webrender::clip_scroll_tree::{{impl}}::add_sticky_frame
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/clip_scroll_tree.rs:350
   7:     0x557da12b0c87 - webrender::frame::{{impl}}::flatten_item
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:716
   8:     0x557da12b1948 - webrender::frame::{{impl}}::flatten_items
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:862
   9:     0x557da12ae727 - webrender::frame::{{impl}}::flatten_stacking_context
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:450
  10:     0x557da12b05fc - webrender::frame::{{impl}}::flatten_item
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:656
  11:     0x557da12b1948 - webrender::frame::{{impl}}::flatten_items
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:862
  12:     0x557da12ae727 - webrender::frame::{{impl}}::flatten_stacking_context
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:450
  13:     0x557da12b05fc - webrender::frame::{{impl}}::flatten_item
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:656
  14:     0x557da12b1948 - webrender::frame::{{impl}}::flatten_items
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:862
  15:     0x557da12b1608 - webrender::frame::{{impl}}::flatten_root
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:831
  16:     0x557da12ad5bd - webrender::frame::{{impl}}::create
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/frame.rs:329
  17:     0x557da12ddc9f - webrender::render_backend::{{impl}}::build_scene
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/render_backend.rs:77
  18:     0x557da12defe9 - webrender::render_backend::{{impl}}::process_document
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/render_backend.rs:251
  19:     0x557da12e05a0 - webrender::render_backend::{{impl}}::run
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/render_backend.rs:429
  20:     0x557da1044de0 - webrender::renderer::{{impl}}::new::{{closure}}
                        at /shared/dev/rust/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/01c38a2/webrender/src/renderer.rs:1464
  21:     0x557da0f8e08a - std::sys_common::backtrace::__rust_begin_short_backtrace<closure,()>
                        at /checkout/src/libstd/sys_common/backtrace.rs:136
  22:     0x557da0f910e3 - std::thread::{{impl}}::spawn::{{closure}}::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:394
  23:     0x557da0f39d4a - std::panic::{{impl}}::call_once<(),closure>
                        at /checkout/src/libstd/panic.rs:296
  24:     0x557da0f92502 - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panicking.rs:480
  25:     0x557da68ff3bc - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  26:     0x557da0f9131c - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /checkout/src/libstd/panicking.rs:459
  27:     0x557da0f90565 - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panic.rs:361
  28:     0x557da0f90ede - std::thread::{{impl}}::spawn::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:393
  29:     0x557da0fc31e3 - alloc::boxed::{{impl}}::call_box<(),closure>
                        at /checkout/src/liballoc/boxed.rs:682
  30:     0x557da68f6f9b - alloc::boxed::{{impl}}::call_once<(),()>
                        at /checkout/src/liballoc/boxed.rs:692
                         - std::sys_common::thread::start_thread
                        at /checkout/src/libstd/sys_common/thread.rs:21
                         - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /checkout/src/libstd/sys/unix/thread.rs:84
  31:     0x7f545508c493 - start_thread
  32:     0x7f5454bb9abe - __clone
  33:                0x0 - <unknown>
ERROR:servo: assertion failed: !self.nodes.contains_key(&id)
```

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #18441 (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18506)

### CODE REVIEW

File: components/layout/display_list_builder.rs [hunk 1]
```diff
@@ -2262,16 +2261,19 @@ impl FragmentDisplayListBuilding for Fragment {
     }
 }
 
-#[derive(Clone, Copy, PartialEq)]
-pub enum EstablishContainingBlock {
-    Yes,
-    No,
+bitflags! {
+    pub flags StackingContextCollectionFlags: u8 {
+        #[doc = "This flow never establishes a containing block."]
```
Comment (emilio) [inline components/layout/display_list_builder.rs]: nit: You can use `///` style comments, if you prefer.

**Review by emilio — APPROVED** [review]
Looks good, r=me.

File: components/layout/display_list_builder.rs (hunk 1 above)
Comment (mrobinson) [reply components/layout/display_list_builder.rs]: Okay. I'll do this. Thanks for the review!

**Comment by mrobinson** [comment]
The build timed out building layout, so I'm going to retry this.

**Pull request merged**